### ISSUE TITLE
Section on Software Management Plans

### ISSUE THREAD

**Issue by c-martinez** [the-turing-way/the-turing-way#2419]
### Summary
Software Management Plans (SMPs) are the software counterpart of Data Management Plans (DMPs). 

In the Netherlands, SMPs are [growing in importance](https://www.esciencecenter.nl/national-guidelines-for-software-management-plans/). There are several existing resources for SMPs:

- [SSI Checklist for a Software Management Plan](https://doi.org/10.5281/zenodo.1422656)
- [Netherlands eScience Center Software Sustainability Protocol](https://doi.org/10.5281/zenodo.1451750)
- [ELIXIR Software Management Plan for Life Sciences](https://doi.org/10.37044/osf.io/k8znb)

We already have a [section on DMPs](https://the-turing-way.netlify.app/reproducible-research/rdm/rdm-dmp.html), but not yet one for SMPs.

### What needs to be done?

- [ ] Write a section on SMPs
- [ ] Start by having a more detailed "What needs to be done" section ;-)

### Who can help?
* @fdiblen ?

---

### Updates
To be continued...

**Comment by EstherPlomp** [comment]
I managed to find some of the TUD resources on this: https://tu-delft-dcc.github.io/docs/software/software_management_plan.html

**Comment by JaroCamphuijsen** [comment]
This might also be an interesting resource for this chapter: https://esciencecenter-digital-skills.github.io/research-software-support/modules/softwaremanagementplans/slides-smp

It is course material, but also useful for self-study.

**Comment by c-martinez** [comment]
Hi,

I've started a [draft](https://github.com/c-martinez/the-turing-way/blob/smp-chapter/book/website/reproducible-research/rdm/rdm-smp.md) for this chapter, with the following structure:

- Introduction
  - What it is?
  - How can they benefit your research?
  - What software to manage?
 - Existing SMP templates
 - Additional Resources

# Questions
1. I wanted to add a section on SMP tools (currently commented out), but I am not sure we have much to add on those lines. We are building an SMP decision tree tool at the eScience Center, but not sure it is mature enough to link to it here. And I don't know if there are many other tools we could include.
1. The DMP chapter has a "A DMP should provide information on six topics" I don't think there are a top 6 or top 10 things to include on an SMP (actually that is the whole point of our guidelines, that which topics should be included depend on the type of software). What do you think?

**Comment by JaroCamphuijsen** [comment]
1. I think it is good to have a section which tells you how to go about SMP's in a hands-on fashion even while we don't have any tools to share. Perhaps a good set of steps can be:
   1. Find out if your institution has its own Software Management Plan template , if so, go to iv.
   2. If not, determine the management level of your software 
   3. Use the SMP template of the appropriate management level from the guide
   4. Answer all questions in the template
5. I know we don't have a top 6, but we do have a top 1, which is "purpose". That should always be there, as software should always be made with a purpose in mind, and its purpose will also determine its management level. It might be good to point to this core requirement and then explain the variety in complexity of software and why there is not one SMP for all.

**Comment by EstherPlomp** [comment]
I agree with @JaroCamphuijsen about the top 1, but perhaps it could be formulated in a top 3? Purpose, sustainability and.....? (I feel you need at least three things to claim that there is a top...). 

Regarding the draft - which already looks very nice! 
- the following sentence comes a bit out of the blue and doesn't tell me a lot: In their [recommendations for Developing and Aligning Policies on Research Software](https://doi.org/10.5281/zenodo.13740998), Science Europe recommend to "Provide guidance regarding research software management".
- It probably needs references to the license and license compatibility sections
- I see some tension between the use of the word developer and researcher. While I understand that everyone who is developing software can be called a developer, I doubt researchers would identify as such (which then later on you also refer to in a paragraph), maybe use both developer and researcher throughout the chapter? 
- I think it would be nice to clarify a bit more what the three types of software are (different categories of research software: analysis code, prototype tools, infrastructure software), perhaps three smaller subsections and the considerations? Maybe I found my top 3 here :)

**Comment by EstherPlomp** [comment]
I read a publication by Grossmann et al that uses some text that could potentially be reused, and it should probably be listed in the resources: Grossmann, Y.V., Lanza, G., Biernacka, K., Hasler, T. and Helbig, K. (2024) Software Management Plans – Current Concepts, Tools, and Application. Data Science Journal, 23: 43, 1–16. DOI: https://doi.org/10.5334/dsj-2024-043

SMP is ..."_instrument to collect ideas and information necessary for their programming activity, which helps them gain awareness and provide for the necessary resources. Such an instrument can be an SMP, which aims to collect all information relevant for the software development in a scientific context to support a structured approach or help distributed teams to adhere to (self-) defined standards"_

_A Software Management Plan (SMP) contains general and technical information about the software project, information on quality assurance, release and public availability as well as legal and ethical aspects that affect the software. The SMP summarises information that adequately describes and documents the creation, documentation, storage, versioning, licensing, archiving and/or publication of the software created or used in a project. Associated hardware and other necessary resources, as well as other associated software and software libraries, text and data publications must also be described and are a special feature of the SMP.
The purpose of an SMP is initially to support the traceability and, if necessary, the longterm usability of the software (for direct application as well as for further processing) and to facilitate user support in the event of queries. The SMP therefore also serves the
purpose of quality assurance (see FAIR for Research Software). The SMP can be linked to one or more DMPs if the software is used for data generation or processing. SMP and DMP can be summarised as output plans._

And I'm not sure if this is already listed: 
_’Checklist for a Software Management Plan’, the first draft for an SMP was created and published there in 2016 (The Software Sustainability Institute, 2018)._

Anyway, not sure if you have the capacity to work on this currently - but happy to continue over the summer or at a later period on this!

**Comment by bakhshir** [comment]
It is a good chapter, @c-martinez . I suggested to add to the introduction part:
* SS Policy of the NLeSC (maybe?)
* Text that Esther has suggested can be merged embedded into the chapter, specially introduction part. 
* the SMP tool is definitely mature enough to link here
* Defining the roles of research software (Version 2) https://doi.org/10.54900/xdh2x-kj281

I think it would be nice to add also after what is it? section one more section:

> ### When and how to write an SMP
> Ideally, an SMP should be drafted during the planning phase of your research project, alongside your DMP. Some funders (e.g., NWO, ZonMw, the Netherlands eScience Center) require an SMP as part of your grant application.
> 
> - Find out if your institution has its own Software Management Plan template , if so, go to iv.
> - If not, determine the management level of your software
> - Use the SMP template of the appropriate management level from the guide
> - Answer all questions in the template
> 
> Some SMP template examples include
> - [SMP template from the Netherlands eScience Center](https://www.esciencecenter.nl/download/51726/?tmstv=1695732347)
> - [WUR SMP template and guidance](https://zenodo.org/records/10473646)
> 
> It should be updated:
> - When you release major versions of your software
> - When you onboard new contributors
> - At the time of publication, to document the final state of the software
> 

Do you want me to make an appropriate rewrite to the chapter?

**Comment by c-martinez** [comment]
Ok, I've edited the [draft ](https://github.com/c-martinez/the-turing-way/blob/smp-chapter/book/website/reproducible-research/rdm/rdm-smp.md)with your comments (@EstherPlomp , @JaroCamphuijsen , @bakhshir -- thanks!).

I've added the "Top 1" suggestion, but it is kind of hidden in "What software to manage?" section.

I'm happy with it as it is (not perfect, but good enough). If you people are also happy with it, let's merge it (and improve it later). If you have any edit suggestions, please make them!

Thanks for all your help so far!

**Comment by bakhshir** [comment]
Looks good @c-martinez One spelling correction "software make explict" -> "software make explicit" ( I don't have edit rights).

**Comment by bakhshir** [comment]
https://tu-delft-dcc.github.io/docs/software/software_management_plan.html Is a dead link now. It is available at https://tu-delft-dcc.github.io/docs/software/fair_software/software_management_plan.html

**Comment by c-martinez** [comment]
Chapter on SMP is now live! Thanks all in this discussion for your contributions. I believe everybody in this discussion is already listed as contributor, but please correct me if I'm wrong.
